[Windows] cannot build as shared library with opencl

Hi,
I was trying to compile elastix with opencl and I got a lot of errors like this:

<details><summary>Compilation output</summary>

```
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUImageToImageFilter.hxx(75,71): error C2491: 'itk::GPUImageToImageFilter<TInputImage,TOutputImage,TParentImag 
eFilter>::GenerateData': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLMovingGenericPyramid\OpenCLMovingG 
enericPyramid.vcxproj]
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUImageToImageFilter.hxx(115,71): error C2491: 'itk::GPUImageToImageFilter<TInputImage,TOutputImage,TParentIma 
geFilter>::GraftOutput': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLMovingGenericPyramid\OpenCLMovingG 
enericPyramid.vcxproj]
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUImageToImageFilter.hxx(150,71): error C2491: 'itk::GPUImageToImageFilter<TInputImage,TOutputImage,TParentIma 
geFilter>::GraftOutput': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLMovingGenericPyramid\OpenCLMovingG 
enericPyramid.vcxproj]
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUInPlaceImageFilter.hxx(44,71): error C2491: 'itk::GPUInPlaceImageFilter<TInputImage,TOutputImage,TParentImag 
eFilter>::PrintSelf': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLMovingGenericPyramid\OpenCLMovingGene 
ricPyramid.vcxproj]
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUInPlaceImageFilter.hxx(53,71): error C2491: 'itk::GPUInPlaceImageFilter<TInputImage,TOutputImage,TParentImag 
eFilter>::ReleaseInputs': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLMovingGenericPyramid\OpenCLMoving 
GenericPyramid.vcxproj]
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUInPlaceImageFilter.hxx(62,71): error C2491: 'itk::GPUInPlaceImageFilter<TInputImage,TOutputImage,TParentImag 
eFilter>::AllocateOutputs': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLMovingGenericPyramid\OpenCLMovi 
ngGenericPyramid.vcxproj]
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUUnaryFunctorImageFilter.hxx(45,87): error C2491: 'itk::GPUUnaryFunctorImageFilter<TInputImage,TOutputImage,T 
Function,TParentImageFilter>::GenerateOutputInformation': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLM 
ovingGenericPyramid\OpenCLMovingGenericPyramid.vcxproj]
C:\itk\build\elastix_build\_deps\elx-src\Common\OpenCL\ITKimprovements\itkGPUUnaryFunctorImageFilter.hxx(54,87): error C2491: 'itk::GPUUnaryFunctorImageFilter<TInputImage,TOutputImage,T 
Function,TParentImageFilter>::GPUGenerateData': definition of dllimport function not allowed [C:\itk\build\elastix_build\_deps\elx-build\Components\MovingImagePyramids\OpenCLMovingGener 
icPyramid\OpenCLMovingGenericPyramid.vcxproj]
```
</details>

And I tracked the error down to the interaction between this line:
https://github.com/SuperElastix/elastix/blob/7db2a8338452468358fc6630c8ab8adfde677d22/Common/OpenCL/CMakeLists.txt#L56
and this file:
https://github.com/SuperElastix/elastix/blob/main/Common/OpenCL/ITKimprovements/itkOpenCLExport.h

Essentially, I was building ITK as a shared library,  and so the export took this path in the `itkOpenCLExport.h`
took the first branch in `if !defined(ITKSTATIC)`  but that since `elxOpenCL` is forced to compile as a static library, it should have taken the other branch.

I assume that the reason that it was force compiled as a static library is for performance. If that's the case, then the fix would be just to delete the first branch to see if it's being compiled as shared or static.
If the proper fix is to not force compile it as static, then a secondary fix would be needed to define `ITKOpenCL_EXPORTS`.

Which ever the approach is best, I can submit a PR for it.

--- a/Common/OpenCL/ITKimprovements/itkOpenCLExport.h
+++ b/Common/OpenCL/ITKimprovements/itkOpenCLExport.h
@@ -24,21 +24,13 @@
 // Setup symbol export
 #define ITKOpenCL_HIDDEN ITK_ABI_HIDDEN
 
-#if !defined(ITKSTATIC)
-#  ifdef ITKOpenCL_EXPORTS
-#    define ITKOpenCL_EXPORT ITK_ABI_EXPORT
-#  else
-#    define ITKOpenCL_EXPORT ITK_ABI_IMPORT
-#  endif /* ITKOpenCL_EXPORTS */
-#else
 /* ITKOpenCL is build as a static lib */
-#  if __GNUC__ >= 4
+#if __GNUC__ >= 4
 // Don't hide symbols in the static ITKOpenCL library in case
 // -fvisibility=hidden is used
-#    define ITKOpenCL_EXPORT ITK_ABI_EXPORT
-#  else
-#    define ITKOpenCL_EXPORT
-#  endif
+#  define ITKOpenCL_EXPORT ITK_ABI_EXPORT
+#else
+#  define ITKOpenCL_EXPORT
 #endif
 
 #endif /* itkOpenCLExport_h */

